src/v1/internal/util.js

```diff
@@ -39,8 +39,29 @@ function isEmptyObjectOrNull(obj) {
 }
 
 function isObject(obj) {
-  const type = typeof obj;
-  return type === 'function' || type === 'object' && Boolean(obj);
+  return typeof obj === 'object' && !Array.isArray(obj) && Boolean(obj);
```

Comment:
I don't know an immediate replacement but not sure if `Boolean(obj)` is the correct thing to check here. Here are a couple of console tests;

`isObject(new Date())` returns `true`
`isObject(new Number())` returns `true`

Do you think we can have a better way of checking for actual object types?

Reply:
I will replace `Boolean(obj)` with `obj !== null` which is much less cryptic.

Technically, `Date` and `Number` are objects so maybe it's okay to accept them. We should probably support parameters given like this:

```javascript
class PersonParams {
  constructor(name, age) {
    this.name = name;
    this.age = age;  
  }
}

session.run('CREATE (:Peson {name: $name, age: $age})', new PersonParams('Bob', 42))
```

and that is why I'm not sure how to allow these but disallow `Date` and `Number`.